crates/uv-python/src/installation.rs

```diff
@@ -92,36 +92,48 @@ impl PythonInstallation {
         let request = request.unwrap_or(&PythonRequest::Default);
 
         // Search for the installation
-        match Self::find(request, environments, preference, cache) {
-            Ok(venv) => Ok(venv),
-            // If missing and allowed, perform a fetch
-            Err(Error::MissingPython(err))
```

Comment:
#10908 added a throw of `Error::Discovery` instead of a `PythonNotFound` error (which is mapped to `MissingPython` transparently here) which means we no longer fallback.

Ideally we'd implement #10908 higher up, like here? but couldn't think of a clean way to do that.